code/modules/projectiles/gun_system.dm

```diff
@@ -829,6 +831,8 @@
 /obj/item/weapon/gun/proc/Fire()
 	if(!target || !(gun_user || istype(loc, /obj/machinery/deployable/mounted/sentry)) || !(CHECK_BITFIELD(item_flags, IS_DEPLOYED) || able_to_fire(gun_user)) || windup_checked == WEAPON_WINDUP_CHECKING)
 		return NONE
+	if((gun_fire_angle) && (get_between_angles(Get_Angle(loc, target), dir2angle(loc.dir)) > (gun_fire_angle / 2)))
+		return NONE
```

Comment:
Extra brackets around the gun fire angle, and couldn't this break if a gun is in something else (e.g gun inside the shoulder module inside the armor inside the mob) or if it's on a turf or in nullspace

First one can use get atom on turf? Rest idk

Reply:
removed the extra bracket, used get_turf. Turf is actually the right thing to use since angles take into account pixel offsets.
If you're in null space tho, uh, how the heck are you firing?